### ISSUE TITLE
Audio buffer is not finite everywhere

### ISSUE THREAD

**Issue by ludzeller** [Anjok07/ultimatevocalremovergui#1045]
The problem has been reported a number of times, is there a fix in sight? I am running the tool on a number of short and longer flac files up to 60 minutes.

Audio buffer is not finite everywhere这个怎么解决呀！ · Issue #1007 · Anjok07/ultimatevocalremovergui
https://github.com/Anjok07/ultimatevocalremovergui/issues/1007

Broken Audio Files Error · Issue #621 · Anjok07/ultimatevocalremovergui
https://github.com/Anjok07/ultimatevocalremovergui/issues/621

[Error] Audio buffer is not finite everywhere · Issue #487 · Anjok07/ultimatevocalremovergui
https://github.com/Anjok07/ultimatevocalremovergui/issues/487

ParameterError: "Audio buffer is not finite everywhere" · Issue #454 · Anjok07/ultimatevocalremovergui
https://github.com/Anjok07/ultimatevocalremovergui/issues/454

ParameterError: Audio buffer is not finite everywhere · Issue #393 · Anjok07/ultimatevocalremovergui
https://github.com/Anjok07/ultimatevocalremovergui/issues/393

Last Error Received:

Process: MDX-Net

If this error persists, please contact the developers with the error details.

Raw Error Details:

ParameterError: "Audio buffer is not finite everywhere"
Traceback Error: "
  File "UVR.py", line 6584, in process_start
  File "separate.py", line 487, in seperate
  File "separate.py", line 354, in final_process
  File "separate.py", line 418, in write_audio
  File "separate.py", line 389, in save_with_message
  File "separate.py", line 396, in deverb_vocals
  File "separate.py", line 1388, in vr_denoiser
  File "lib_v5/spec_utils.py", line 340, in cmb_spectrogram_to_wave
  File "librosa/util/decorators.py", line 104, in inner_f
  File "librosa/core/audio.py", line 606, in resample
  File "librosa/util/decorators.py", line 88, in inner_f
  File "librosa/util/utils.py", line 294, in valid_audio
"

Error Time Stamp [2023-12-21 16:12:55]

Full Application Settings:

vr_model: UVR-DeEcho-DeReverb
aggression_setting: 0
window_size: 512
mdx_segment_size: 256
batch_size: Default
crop_size: 256
is_tta: False
is_output_image: False
is_post_process: False
is_high_end_process: False
post_process_threshold: 0.2
vr_voc_inst_secondary_model: No Model Selected
vr_other_secondary_model: No Model Selected
vr_bass_secondary_model: No Model Selected
vr_drums_secondary_model: No Model Selected
vr_is_secondary_model_activate: False
vr_voc_inst_secondary_model_scale: 0.9
vr_other_secondary_model_scale: 0.7
vr_bass_secondary_model_scale: 0.5
vr_drums_secondary_model_scale: 0.5
demucs_model: Choose Model
segment: Default
overlap: 0.25
overlap_mdx: Default
overlap_mdx23: 8
shifts: 2
chunks_demucs: Auto
margin_demucs: 44100
is_chunk_demucs: False
is_chunk_mdxnet: False
is_primary_stem_only_Demucs: False
is_secondary_stem_only_Demucs: False
is_split_mode: True
is_demucs_combine_stems: True
is_mdx23_combine_stems: True
demucs_voc_inst_secondary_model: No Model Selected
demucs_other_secondary_model: No Model Selected
demucs_bass_secondary_model: No Model Selected
demucs_drums_secondary_model: No Model Selected
demucs_is_secondary_model_activate: False
demucs_voc_inst_secondary_model_scale: 0.9
demucs_other_secondary_model_scale: 0.7
demucs_bass_secondary_model_scale: 0.5
demucs_drums_secondary_model_scale: 0.5
demucs_pre_proc_model: No Model Selected
is_demucs_pre_proc_model_activate: False
is_demucs_pre_proc_model_inst_mix: False
mdx_net_model: UVR-MDX-NET Inst HQ 3
chunks: Auto
margin: 44100
compensate: Auto
denoise_option: None
is_match_frequency_pitch: True
phase_option: Automatic
phase_shifts: None
is_save_align: False
is_match_silence: True
is_spec_match: False
is_mdx_c_seg_def: False
is_invert_spec: False
is_deverb_vocals: True
deverb_vocal_opt: All Vocal Types
voc_split_save_opt: Lead Only
is_mixer_mode: False
mdx_batch_size: Default
mdx_voc_inst_secondary_model: VR Arc: 1_HP-UVR
mdx_other_secondary_model: No Model Selected
mdx_bass_secondary_model: No Model Selected
mdx_drums_secondary_model: No Model Selected
mdx_is_secondary_model_activate: False
mdx_voc_inst_secondary_model_scale: 0.9
mdx_other_secondary_model_scale: 0.7
mdx_bass_secondary_model_scale: 0.5
mdx_drums_secondary_model_scale: 0.5
is_save_all_outputs_ensemble: True
is_append_ensemble_name: False
chosen_audio_tool: Manual Ensemble
choose_algorithm: Min Spec
time_stretch_rate: 2.0
pitch_rate: 2.0
is_time_correction: True
is_gpu_conversion: True
is_primary_stem_only: False
is_secondary_stem_only: True
is_testing_audio: False
is_auto_update_model_params: True
is_add_model_name: False
is_accept_any_input: False
is_task_complete: False
is_normalization: True
is_wav_ensemble: False
is_create_model_folder: False
mp3_bit_set: 320k
semitone_shift: 0
save_format: FLAC
wav_type_set: PCM_16
cuda_set: Default
help_hints_var: True
set_vocal_splitter: No Model Selected
is_set_vocal_splitter: False
is_save_inst_set_vocal_splitter: False
model_sample_mode: False
model_sample_mode_duration: 30
demucs_stems: All Stems
mdx_stems: All Stems

**Comment by ludzeller** [comment]
I converted the input flac to wav and turned off the vocal splitter auto deverbing and then it ran through nicely.

Still, I encountered more problems when attempting to batch deverb the resulting files, also introducing new issues such as silent or even fully glitched outputs. Has it to do with the VR Architecture somehow, were the deverbing is happening, so that that produces invalid buffers at time leading to all kinds of unwanted behaviour downstream in file compression and saving?

**Comment by ChrisTG742** [comment]
To the coders: Maybe this here could be helpful solving the issue: https://stackoverflow.com/questions/57097840/librosa-parametererror-audio-buffer-is-not-finite-everywhere-cutting-wav-files

**Comment by ludzeller** [comment]
Yes, it seems like there are NaNs in the buffer, but the question then is where do they come from and why?

**Comment by ChrisTG742** [comment]
There is one hint about the NaNs in the SO-article I linked. They say it is connected to the usage of integers instead of floats.
@Anjok07 is there any type-conversion from float to int in your code?

**Comment by ChrisTG742** [comment]
Interestingly, the file "lib_v5/spec_utils.py" is completely missing in my installation here (Windows installer), though it runs without any problems or errors.
@Anjok07 could it be that the code-base of the Windows-installer differs from what there's currently in your repo?

**Comment by Abedalhkeem-z** [comment]
Any updates on solving the problem?

**Comment by ANDYVDL** [comment]
This is still an issue
